src/stores/businessStore.ts

```diff
@@ -91,9 +91,23 @@ export const useBusinessStore = defineStore('business', {
       return state.businessInfo.lastDirectorChangeDate
     },
 
-    /** The legal name. */
+    /** The legal name or alternate name if is firm. */
     getLegalName (state: BusinessStateIF): string {
-      return state.businessInfo.legalName
+      if (!GetFeatureFlag('enable-legal-name-fix')) {
+        return state.businessInfo.legalName
+      }
+      if (this.isFirm) {
+        return this.getAlternateName || 'Unknown'
```

Comment:
The unknown here is just a failsafe value. I confirmed with Argus that firms should and will always have the alternate names array. 

Line 102 is same thing. We should always return legal name in case not firm. That also should always be present.